packages/nhost_sdk/lib/src/auth.dart

```diff
@@ -479,24 +520,28 @@ class AuthClient {
         },
         responseDeserializer: Session.fromJson,
       );
-    } on ApiException catch (e, st) {
-      if (e.statusCode == unauthorizedStatus) {
+
+      await setSession(res!);
+      _sessionCompleter?.complete(res);
+      return res;
+    } on Exception catch (e, st) {
+      if (e is ApiException && e.statusCode == unauthorizedStatus) {
         log.finest('Unauthorized refresh token. Forcing signout.');
         await signOut();
-        return;
-      } else {
-        log.severe('API exception during token refresh', e, st);
-        return;
       }
-    } catch (e, st) {
+
       log.severe('Exception during token refresh', e, st);
-      return;
+      _sessionCompleter?.completeError(e, st);
+
+      // Inform subscribers of the failure. If there are none, rethrow the
```

Comment:
nit: The comment is misleading. It will rethrow regardless if there are subscribers of the failure or not.

Reply:
Ah, thank you. Will correct.